charmhelpers/contrib/storage/linux/ceph.py

```diff
@@ -645,11 +647,20 @@ def __init__(self, service, name=None, pg_num=None, replicas=None,
             # we will fail with KeyError if it is not provided.
             self.replicas = op['replicas']
             self.pg_num = op.get('pg_num')
+            self.profile_name = op.get('crush-profile')
```

Comment:
You may want to do this instead?

```python
self.profile_name = op.get('crush-profile', profile_name or 'replicated_rule')
```

i.e. always ensure that the profile name exists even if it's not in the `op`; otherwise 'older' users of the library might not have updated their 'ops' to use the `crush-profile` key?